File: Webfuel.Api/Api/SupportTeam/SupportTeamUserApi.cs
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Webfuel.Domain;
using Webfuel.Domain.StaticData;

namespace Webfuel.App
{
    [ApiService]
    public static class SupportTeamUserApi
    {
        public static void RegisterEndpoints(IEndpointRouteBuilder app)
        {
            // Commands

            app.MapPost("api/support-team-user/insert", Insert)
                .RequireIdentity();

            app.MapPost("api/support-team-user/delete", Delete)
                .RequireIdentity();

            // Querys

            app.MapPost("api/support-team-user/query", Query)
                .RequireIdentity();
        }

        public static Task Insert([FromBody] InsertSupportTeamUser command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task Delete([FromBody] DeleteSupportTeamUser command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task<QueryResult<SupportTeamUser>> Query([FromBody] QuerySupportTeamUser command, IMediator mediator)
        {
            return mediator.Send(command);
        }
    }
}

File: Webfuel.App/ClientApp/src/app/api/support-team-user.api.ts
import { EventEmitter, Injectable, inject } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { ApiService, ApiOptions } from '../core/api.service';
import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from '@angular/router';
import { IDataSource } from 'shared/common/data-source';
import { InsertSupportTeamUser, DeleteSupportTeamUser, QuerySupportTeamUser, QueryResult, SupportTeamUser } from './api.types';

@Injectable()
export class SupportTeamUserApi {
    constructor(private apiService: ApiService) { }
    
    public insert (body: InsertSupportTeamUser, options?: ApiOptions): Observable<any> {
        return this.apiService.request<InsertSupportTeamUser, any>("POST", "api/support-team-user/insert", body, options);
    }
    
    public delete (body: DeleteSupportTeamUser, options?: ApiOptions): Observable<any> {
        return this.apiService.request<DeleteSupportTeamUser, any>("POST", "api/support-team-user/delete", body, options);
    }
    
    public query (body: QuerySupportTeamUser, options?: ApiOptions): Observable<QueryResult<SupportTeamUser>> {
        return this.apiService.request<QuerySupportTeamUser, QueryResult<SupportTeamUser>>("POST", "api/support-team-user/query", body, options);
    }
}

